home/modules/alerts/pages/cve-2023-22949.adoc

```diff
@@ -0,0 +1,40 @@
+:pp: {plus}{plus}
+
+Type: Security
+
+Original Date: January 2024
+
+== Summary
+In April 2023, link:https://nvd.nist.gov/vuln/detail/CVE-2023-22949[NVD - CVE-2023-22949] was filed as a vulnerability with a medium base score of 4.9, pertaining to TigerGraph version 3.7.0 and TigerGraph Cloud.
+The CVE notes that authenticated GSQL access requests were logged by TigerGraph, including both the username and password of the user in an easily decodable base64 form.
+That could allow a TigerGraph administrator to effectively harvest usernames/passwords.
+
+*TigerGraph addressed this vulnerability in February 2023, prior to the filing of the CVE*, in the following ways:
+
+* A bug fix was applied in all production (LTS) branches (3.6 and 3.9).
+* The 3.7.0 Feature Preview branch has now reached end of life and is no longer available.
+* TigerGraph Cloud tracks with the most recent database updates, ensuring the most secure and stable versions of TigerGraph on the platform.
+
+== Remediation Details
+TigerGraph detected this vulnerability prior to the filing of the CVE.
+The vulnerability was fully patched by always masking user credentials in log files, as highlighted in the example below:
+
+* Before the fix, the log was: `"auth":"Basic dGlnZXJncmFwaDp0aWdlcmdyYXBo"`.
+* After the fix, the log is: `"auth":<masked>`.
+
+Notably, TigerGraph’s two LTS (Long Term Support) branches, 3.6 and 3.9, were patched in February 2023 (GLE-4774 and GLE-4718), two months before this CVE.
+The CVE detected the vulnerability in 3.7.0, a Feature Preview branch released in September 2022.
+The 3.7.0 Feature Preview branch has now reached end of life and is no longer available.
+Additionally, the CVE detected the vulnerability in TigerGraph Cloud, but TigerGraph Cloud tracks with the most recent database updates, fixing the bug by ensuring the most secure and stable versions are utilized on the platform.
+
+The TigerGraph platform has undergone significant security enhancements. In past versions, user credentials, including those of the administrative user, were inadvertently logged in platform files during HTTP authentications.
+As this potential risk was addressed and rectified it showcases the platform's commitment to robust security measures.
+The impact of this vulnerability bug was effectively eliminated in updated distributed versions of TigerGraph.
+
+The bug fix not only fortified the platform, but also strengthened TigerGraph's security posture further against other exploits, such as link:https://graphsql.atlassian.net/wiki/spaces/Engineerin/pages/3169255425/Response+to+CVE-2023-22949?focusedCommentId=3182068225[CVE-2022-30331] (which was also disputed in TigerGraph's xref:cve-2022-30331.adoc[documentation] in a previous alert), by removing this attack vector for obtaining administrative user login credentials.
+
+TigerGraph versions 3.6.3, 3.9.0 and later and TigerGraph Cloud have already undergone these necessary updates.
```

Comment:
I'm not sure what "these" changes are, the ones for for CVE (CVE-2023-22949), or some of the other security improvements.  Also, it is not helpful for us to characterize them as "necessary".

Reply:
Thank you. fixed this with latest commit.